src/app_service/service/contacts/service.nim

```diff
@@ -349,7 +376,7 @@ QtObject:
       return self.contacts[pubkey]
 
     result = self.fetchContact(pubkey)
```

Comment:
I just realized that we may hit the DB here, so it's a possible case where we could have a freeze. That may explain why we still see freezes when switching.

No need to fix here, but something to consider.

Reply:
Haven't seen that as a culprit in profiled data, although yes, we should make all potential DB hits async. I would leave it as is for now, otherwise it will complicate logic of getting contact details a lot.